src/sentry/static/sentry/app/views/eventsV2/table/draggableColumns.tsx

```diff
@@ -0,0 +1,304 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import styled from 'react-emotion';
+
+import space from 'app/styles/space';
+import {
+  UserSelectValues,
+  setBodyUserSelect,
+} from 'app/components/events/interfaces/spans/utils';
+
+import {TableColumn} from './types';
+
+export type DraggableColumnsChildrenProps = {
+  startColumnDrag: (
+    event: React.MouseEvent<SVGSVGElement, MouseEvent>,
+    initialColumnIndex: number
+  ) => void;
+  draggingColumnIndex: undefined | number;
+  destinationColumnIndex: undefined | number;
+};
+
+type Props = {
+  children: (props: DraggableColumnsChildrenProps) => JSX.Element;
+  columnOrder: TableColumn<React.ReactText>[];
+  onDragDone: (props: {
+    draggingColumnIndex: undefined | number;
+    destinationColumnIndex: undefined | number;
+  }) => void;
+};
+
+type State = {
+  isDragging: boolean;
+  left: undefined | number;
+  top: undefined | number;
+
+  draggingColumnIndex: undefined | number;
+  destinationColumnIndex: undefined | number;
+};
+
+class DraggableColumns extends React.Component<Props, State> {
+  state: State = {
+    isDragging: false,
+
+    // initial coordinates for when the drag began
+    left: void 0,
+    top: void 0,
+
+    draggingColumnIndex: void 0,
+    destinationColumnIndex: void 0,
+  };
+
+  previousUserSelect: UserSelectValues | null = null;
+  portal: HTMLElement | null = null;
+  dragGhostRef = React.createRef<HTMLDivElement>();
+
+  startColumnDrag = (
+    event: React.MouseEvent<SVGSVGElement, MouseEvent>,
+    initialColumnIndex: number
+  ) => {
+    const isDragging = this.state.isDragging;
+
+    if (isDragging || event.type !== 'mousedown') {
+      return;
+    }
+
+    // prevent the user from selecting things outside the minimap when dragging
+    // the mouse cursor outside the minimap
+
+    this.previousUserSelect = setBodyUserSelect({
+      userSelect: 'none',
+      MozUserSelect: 'none',
+      msUserSelect: 'none',
+    });
+
+    // attach event listeners so that the mouse cursor can drag anywhere
+
+    window.addEventListener('mousemove', this.onDragMove);
+    window.addEventListener('mouseup', this.onDragEnd);
+
+    const columnBeingDragged = this.props.columnOrder[initialColumnIndex];
+
+    this.setState({
+      isDragging: true,
+      draggingColumnIndex: initialColumnIndex,
+      left: event.pageX,
+      top: event.pageY,
+    });
+
+    console.log('dragging column', initialColumnIndex, columnBeingDragged);
+
+    // TODO: experiment, clean this up
+
+    const foo = document.querySelectorAll('.grid-head-cell-button');
+    const moo = Array.from(foo).map(column => {
+      return column.getBoundingClientRect();
+    });
+
+    console.log('foo', foo);
+    console.log('moo', moo);
+
+    foo.forEach(element => {
+      const rects = element.getBoundingClientRect();
+
+      const reference = document.createElement('div');
```

Comment:
Would this be a good place to use a portal to render the ghost?

Reply:
These were DOM elements I've created for debugging; I've since removed them.